docs/CToSwiftNameTranslation.md

```diff
@@ -430,4 +453,86 @@ Although enumerators always have global scope in C, they are often imported as m
+
+```objc
+struct SPKSpaceflightBooking {
+  const SPKLocation * _Nullable destination;
+  bool roundTrip __attribute__((swift_name("isRoundTrip")));
+};
+```
+
+```swift
+struct SPKSpaceflightBooking {
+  var destination: UnsafePointer<SPKLocation>?
+  var isRoundTrip: Bool
+}
+```
+
+
+### Objective-C methods
+
+The `swift_name` attribute can be used to give an Objective-C method a custom name. The value of the attribute must be a full Swift function name, including parameter labels.
+
+```objc
+- (void)doSomethingToFoo:(Foo *)foo bar:(int)bar
+  __attribute__((swift_name("doSomethingImportant(to:bar:)")));
+```
+
+```swift
+func doSomethingImportant(to foo: UnsafeMutablePointer<Foo>, bar: Int32)
+```
+
+As with functions, an underscore can be used to represent an empty parameter label.
+
+Methods that follow the NSError out-parameter convention may provide one fewer parameter label than the number of parameters in the original method to indicate that a parameter should be dropped, but they do not have to. The `swift_error` attribute is still respected even when using a custom name for purposes of transforming an NSError out-parameter and the method return type.
+
+```objc
+- (BOOL)doSomethingRiskyAndReturnError:(NSError **)error
+  __attribute__((swift_name("doSomethingRisky()")));
+- (BOOL)doSomethingContrived:(NSString *)action error:(NSError **)outError
+  __attribute__((swift_name("doSomethingContrived(_:error:)")));
+```
+
+```swift
+func doSomethingRisky() throws
+func doSomethingContrived(_ action: String, error: ()) throws
+```
+
+A base name of "init" can be used on a *class* method that returns `instancetype` or the containing static type in order to import that method as an initializer. Any other custom name *prevents* a class method from being imported as an initializer even if it would normally be inferred as one.
+
+```objc
++ (Action *)makeActionWithHandler:(void(^)(void))handler
+  __attribute__((swift_name("init(handler:)")));
++ (instancetype)makeActionWithName:(NSString *)name
+  __attribute__((swift_name("init(name:)")));
+```
+
+```swift
+/* non-inherited */ init(handler: () -> Void)
+init(name: String)
+```
+
+A no-argument method imported as an initializer can be given a dummy argument label to disambiguate it from the no-argument `init()`, whether the method is an init-family instance method or a factory class method in Objective-C.
+
+```objc
+- (instancetype)initSafely
+  __attribute__((swift_name("init(safe:)")));
++ (instancetype)makeDefaultAction
+  __attribute__((swift_name("init(default:)")));
+```
+
+```swift
+init(safe: ())
+init(default: ())
+```
+
+A custom name on a method with one of Objective-C's subscript selectors (`objectAtIndexedSubscript:`, `objectForKeyedSubscript:`, `setObject:atIndexedSubscript:`, or `setObject:forKeyedSubscript:`) prevents that method from being imported as a subscript or used as the accessor for another subscript.
```

Comment:
I think this is the only discussion of subscripts so far in this document?

It's worth noting somewhere (maybe not here) that only instance methods are eligible to be subscripts. This is because, although Objective-C sort of tries to allow subscripts on class methods, there's no actual way to call them as subscripts.

Reply:
> I think this is the only discussion of subscripts so far in this document?

Yeah, I haven't actually done the *regular* section on ObjC methods yet. It's the hardest section because of omit-needless-words, so I've been putting it off.

> although Objective-C sort of tries to allow subscripts on class methods

I'm not quite sure what this means when it doesn't work, but sure. I think I'll just throw in the word "instance".

Reply:
What I mean is that you can define an `+objectAtIndexedSubscript:`, but the instance you would call it on is just going to be `Class` because ObjC is never more specific than that, so you can't use subscript syntax with it in ObjC.

Reply:
`Class` behaves like `id`, so it ought to work as long as *anyone* has declared a class subscript accessor. Also literally referencing the class ought to work (`NSFoo[0]`), the way it does for properties (even before class properties were a thing).

Reply:
I don't think you can use subscripts on `id`, either. I think I actually wrote some ObjC code that tried to declare and use a class subscript and it didn't work.

Reply:
No, that one works.

```objc
@import Foundation;

void test(id foo) {
  (void)foo[0];
}
```